### ISSUE TITLE
Vulnerability dashboard shows solved vulnerabilities (Manjaro - agent)

### ISSUE THREAD

**Issue by blachot** [wazuh/wazuh#23605]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| Wazuh agent 4.7.4 | Vulnerability | Agent | Sources | Manjaro |

Hello,
Wazuh Manager is on same version. It has been installed with AWS Wazuh AMI image on Amazon Linux 2.
Agents running with OS Windows 11, Amazon Linux 2 or Amazon Linux 2023 work. properly
But I have problem with agents running with Manjaro OS. I see in the vulnerability dashboard vulnerability entries for the CVEs where the corresponding events have "Solved" status.

```
 /var/ossec/bin/agent_control -i 3955

Wazuh agent_control. Agent information:
   Agent ID:   3955
   Agent Name: BLT-LAPTOP
   IP address: any
   Status:     Active

   Operating system:    Linux |bruno-vostro157510 |6.6.26-1-MANJARO |#1 SMP PREEMPT_DYNAMIC Wed Apr 10 20:11:08 UTC 2024 |x86_64
   Client version:      Wazuh v4.7.4
   Configuration hash:  1c62605d6a7c6e22748693b1fa67d754
   Shared file hash:    9bebd9788c460073da0e944ca033aec7
   Last keep alive:     1716487383

   Syscheck last started at:  Thu May 23 16:54:14 2024
   Syscheck last ended at:    Thu May 23 16:54:17 2024
```

![Wazuh vulnerability dashbord](https://github.com/wazuh/wazuh/assets/58725936/fbda6884-23db-41ed-be7d-5282080d6cc2)

![Wazuh events CVE-2022-2068](https://github.com/wazuh/wazuh/assets/58725936/136eadf2-435c-43d1-a061-b3557bc7f196)
![Wazuh events CVE-2021-3773](https://github.com/wazuh/wazuh/assets/58725936/e0405d40-ad83-4f2b-b2a3-48a0f7a2f716)
![Wazuh events CVE-2023-45853](https://github.com/wazuh/wazuh/assets/58725936/e5e251b3-be8f-44d9-a134-63f9a2cc00a3)

The problem is also present for other vulnerabilities.

The problem has persisted for several months now, despite the switch to the latest version.

I seen that the [same issue](https://github.com/wazuh/wazuh/issues/20318) has been closed without solution for me.

Thank you for your help.

Regards.

**Comment by rmsteffen** [comment]
I am also having this issue.  Vulnerabilities show as solved in the Events but are still showing up on the dashboard.  Is there a solution for this?

**Comment by pereyra-m** [comment]
Hello @blachot !

The events tab shows alerts related to changes in the vulnerability detector's inventory, but don't reflect the current state of the agent.
Every time the scanner finds a vulnerable package, an alert is generated. The same happens when a package is removed/upgraded, the "solved" alert is sent.

When you upgrade a package to a vulnerable version, for example, it'll still appear in the inventory. The "solved" alerts will be generated because the upgrade is seen as a removal and an installation.
We can analyze the reported vulnerabilities if you share the full agent's package inventory, to make sure they aren't false positives.

Regards.

**Comment by rmsteffen** [comment]
I'm having the same thing and happy to share if you tell me what you need me to run.  In my case the dashboard shows vulns (there indeed were vulns, I applied a bunch of updates, then in the events I saw the Solved events, but the Dashboard continued to show vulns, no further events after 'solved' ones appeared).  This is on Windows not Manjaro however.  If I remove the agent (delete it in manager, remove it on the client PC including the ossec agent directory in program files) and then re-install it, it will come up as having no vulnerabilities and everything is fine.  But I think there is a bit somewhere that doesn't update the dashboard, happy to dump any log or screenshot or whatever if that helps.  Somewhere else someone mentioned making custom alerts and raising the severity of the Solved alerts to 7 from 3 fixes the issue, which seemed odd?

**Comment by blachot** [comment]
Hello @pereyra-m 

Thank you for your reply.

To start from scratch, I stopped the agent, deleted it on the manager, updated the Manjaro OS and restarted the agent.
This time, I no longer have a “solved” event, but from what you say, that's not very important.
I've attached the export of the vulnerabilities. I don't know if you wanted more information.

But as an example, I don't understand the problem on openssl fixed in version 3.0.4 since June 2022 https://www.openssl.org/news/vulnerabilities-3.0.html#CVE-2022-2068. Maybe it's due to the "unknown" status on https://security.archlinux.org/AVG-2765
In any case, this problem is annoying because Wazuh cannot be relied upon to manage vulnerabilities in the context of ISO 27001 certification.
Let me know if you need any further information to help us.

For your information, I don't encounter this problem on the other operating systems tested (Windows 11, Amazon Linux 2 and Amazon Linux 2023).

Regards.

**Comment by blachot** [comment]
[vulnerabilities.csv](https://github.com/user-attachments/files/15523644/vulnerabilities.csv)
Here is export of vu

**Comment by pereyra-m** [comment]
Hello again @blachot , thank you for the information provided.

First of all, let me say that the vulnerability detector module has been fully refactored and it will be available in Wazuh v4.8.0. This type of issue will be handled easily with the changes implemented because the feed will be sanitized by our team.

In the example you gave of OpenSSL, the Arch Linux feed doesn't provide a fixed version for this CVE, so it considers the package vulnerable no matter what version you have. Every time you perform an upgrade of that package, two alerts will be generated:
- A solved alert, because it was removed
- An active alert, because the new version is installed

There might be a slight delay in the processing of the events, causing the alerts to arrive in different order. 
And a similar situation might happen with the rest of the CVEs, the vendor feed isn't accurate about the version range.


@rmsteffen 
About this

> Somewhere else someone mentioned making custom alerts and raising the severity of the Solved alerts to 7 from 3 fixes the issue, which seemed odd?

This is just a workaround that reduces the level of an alert of a specific package to avoid receiving false positive alerts. 
But it isn't a fix and the CVE will be in the inventory anyway. It'd be valuable if you could upload the vulnerabilities detected, the list of KBs, and the packages' inventory in a different issue so we can analyze them.

Finally, I'd suggest for both that you test the new module that will be released soon. Your feedback is important to adjust the efficacy of the vulnerability detector. I apologize for the inconvenience.

Regards.
